src/c_api/c_api.cc

```diff
@@ -114,13 +114,19 @@ void CustomFComputeDispatcher(const std::string op_name,
                               const std::vector<OpReqType>& req,
                               const std::vector<NDArray>& outputs) {
   std::vector<void*> in_data, out_data;
-  std::vector<const int64_t *> in_shapes, out_shapes;
+  std::vector<const int64_t*> in_shapes, out_shapes;
   std::vector<int> in_dims, out_dims;
   std::vector<int> in_types, out_types;
   std::vector<size_t> in_verIDs, out_verIDs;
   std::vector<const char*> in_dev_type, out_dev_type;
   std::vector<int> in_dev_id, out_dev_id;
 
+  // Extra data for sparse inputs.
+  std::vector<void*> in_indices;
+  std::vector<void*> in_indptr;
+  std::vector<int64_t> in_indices_shapes;
+  std::vector<int64_t> in_indptr_shapes;
```

Comment:
lets just initialize these vectors to inputs.size(). Then it'll be easier to index into them with the same index as the tensor input

Reply:
Done.